Source/Dafny/Resolver.cs

```diff
@@ -12501,6 +12494,48 @@ private class RBranchExpr : RBranch {
       }
     }
 
+    private ExtendedPattern RemoveDisjunctivePatterns(ExtendedPattern pat, bool allowVariables) {
```

Comment:
The name only captures half of the behavior, maybe rename to `HandleIllegalPatterns` (Handle over remove since you are both replacing them and emitting an error)

Reply:
Good point